doc/case_studies/numpy-tooltip-images.rst

```diff
@@ -0,0 +1,195 @@
+when the dots are hovered with the mouse.
+We can see that the large white blobs
+correspond to the higher area measurements
+as expected.
+
+.. altair-plot::
+    import altair as alt
+
+    # The random() function is used to jitter points in the x-direction
+    alt.Chart(df_plot, width=alt.Step(40)).mark_circle(xOffset=alt.expr('random() * 16 - 8')).encode(
+        x='group',
+        y=alt.Y(alt.repeat(), type='quantitative'),
+        tooltip=['image'],
+        color='group',
+    ).repeat(
+        ['image1_area', 'image2_area']
+    ).resolve_scale(
+        y='shared'
+    ).properties(
+        title='Comparison of blob areas'
+    )
+
+If we want to have even more fun and get a bit more sophisticated,
+we could show one chart at a time
+and update what is shown on the y-axis
+as well as what is shown in the image tooltip
+based on a dropdown selector.
+We start by defining a tooltip that only contains a single image
+instead of both the images concatenated together.
+
+.. altair-plot::
+    :output: repr
+
+    def create_tooltip_image(img_arr):
+        """Rescale and convert an image to a base64 string."""
+        # print(img_arr)
+        # Create a PIL image from the array.
+        # Multiplying by 255 and recasting as uint8 for the images to occupy the entire supported instensity space from 0-255
+        img = Image.fromarray((255 * img_arr).astype('uint8'))
+
+        # Convert to base64 encoded image string that can be displayed in the tooltip
+        buffered = BytesIO()
+        img.save(buffered, format="PNG")
+        img_str = base64.b64encode(buffered.getvalue()).decode()
+        return f"data:image/png;base64,{img_str}"
+
+    # The column with the base64 image string must be called "image" in order for it to trigger the image rendering in the tooltip
+    df[['image1_base64', 'image2_base64']] = df[['image1', 'image2']].applymap(create_tooltip_image)
+    # Dropping the image arrays since they are large an no longer needed
+    # Also drop the previous tooltip image for clarity
+    df_plot = df.drop(columns=['image1', 'image2', 'image'])
+    df_plot
+
+In our chart,
+we need to use a transform to update
+both the y-axis column as well as the tooltip column
+dynamically based on the selection in the dropdown.
+The comments in the code explain more in detail what each line
+in this chart specification does.
+
+.. altair-plot::
+    metric_dropdown = alt.binding_select(
+        options=['image1_area', 'image2_area'],
+        name='Image metric '
+    )
+    metric_param = alt.param(
+        value='image1_area',
+        bind=metric_dropdown
+    )
+    alt.hconcat(
+        # This first chart is the axis title and is only needed because
+        # Vega-Lite does not yet support passing an expression directly to the axis title
+        alt.Chart().mark_text(angle=270, dx=-150, fontWeight='bold').encode(
+            alt.TextValue(alt.expr(f'{metric_param.name}'))
```

Comment:
Interestingly, sphinx does not allow us to reference the actual name of the `metric_param` as a string, which works in jupyterlab and the vega editor. The only way around it I found was to use an f-string with `metric_param.name` which is what we use elsewhere in the docs and it's probably more convenient anyways since we don't need to manually set a name for the param.

Reply:
Hmm weird, can't explain why that happens but I agree that using `.name` is more convenient.